packages/ses/src/compartment-shim.js

```diff
@@ -83,13 +89,143 @@ const assertModuleHooks = compartment => {
   }
 };
 
-/**
- * Compartment()
- * The Compartment constructor is a global. A host that wants to execute
- * code in a context bound to a new global creates a new compartment.
+/** SharedCompartmentConstructor
+ *
+ * All compartments share a prototype, though there are many compartment
+ * constructors.
+ * The compartment shim introduces a fully powered global Compartment constructor.
+ * The lockdown shim replaces that Compartment constructor with one whose descendants
+ * all share the new tamed and hardened environment.
+ * Within each Compartment, there is a unique Compartment constructor that
+ * closes over the parent.
+ *
+ * Before lockdown, we expect every constructor to have a prototype and for the
+ * prototype to refer back to the constructor.
+ * After lockdown, we expect the constructors from all compartments to share a
+ * prototype, but the prototype can no longer carry a reference back to any
+ * particular constructor.
+ * For example, every compartment has a unique and powerful Function
+ * constructor, each of these has the same prototype.
+ * Because the prototype is shared among compartments, the shared constructor
+ * on the prototype must be powerless.
+ *
+ * Following this pattern, the shared compartment constructor is also powerless
+ * and simply throws an exception.
+ */
+export const SharedCompartmentConstructor = function Compartment() {
+  throw new TypeError(
+    `Cannot call or construct compartment with shared constructor`,
+  );
+};
+
+SharedCompartmentConstructor.toString = () => {
+  return 'function Compartment() { [shim code] }';
+};
+
+/** CompartmentPrototype
+ * Every compartment constructor shares this prototype.
+ * As with other primordials, the
  */
-export class Compartment {
-  constructor(endowments = {}, modules = {}, options = {}) {
+const CompartmentPrototype = {
+  constructor: SharedCompartmentConstructor,
+
+  get globalThis() {
+    return privateFields.get(this).globalObject;
+  },
+
+  /**
+   * @param {string} source is a JavaScript program grammar construction.
+   * @param {{
+   *   transforms: Array<Transform>,
+   *   sloppyGlobalsMode: bool,
```

Comment:
We need to remember to talk to Moddable (attn @phoddie ) and @bmeck about `sloppyGlobalsMode`. I am reluctant to propose this, but I don't know what to do without it.

Reply:
Narrator: we didn’t remember.

Captured this in an agenda doc and shared.